pdns/auth-catalogzone.hh

```diff
@@ -43,8 +43,8 @@ public:
 
   static constexpr const char* getTypeString(enum CatalogType type)
   {
-    const char* types[] = {"none", "producer", "consumer"};
-    return types[type];
+    const std::array<const char*, 3> types = {"none", "producer", "consumer"};
+    return types.at(type);
   }
```

Comment:
If it makes sense I can move this change in an other PR. Hesitated to change to string_view should I do it?

Reply:
what's the reason you suggest `string_view` instead of just `string`?

Reply:
hmm it's my limit with C++ memory model, I don't know now (exactly) where are stored `static` (program lived) `strings` in cpp. At least I know that a `char *` is not heap allocated in this context.

Context: I mostly do rust, `String` are heap beast where `&'static str` are in memory image of the program.

Reply:
In this case it would be a pain would need change in json11 maybe not today. I suppose.

```cpp
auth-catalogzone.cc: In member function ‘void CatalogInfo::fromJson(const std::string&, CatalogType)’:
auth-catalogzone.cc:41:15: error: no match for ‘operator[]’ (operand types are ‘json11::Json’ and ‘std::string_view’ {aka ‘std::basic_string_view<char>’})
   41 |     if (!d_doc[getTypeString(d_type)].is_null()) {
      |               ^
```

Reply:
> hmm it's my limit with C++ memory model, I don't know now (exactly) where are stored `static` (program lived) `strings` in cpp. At least I know that a `char *` is not heap allocated in this context.
> 
> Context: I mostly do rust, `String` are heap beast where `&'static str` are in memory image of the program.

Often `string` implementations store the data for small string inside the object itself, and only use the heap for larger strings. For `const string`s there might be more smarter things going on.

But does it really matter how strings are stored exactly?  The main thing is to avoid naked pointers. 

As your json11 example shows, many functions expect `string` and not `string_view`. Another reason to go for `string`.

Reply:
It's more clear, will just goes for `string` , thanks for the explanation.